### PR TITLE
Nodelinktest

### PR DESCRIPTION
**Pull request by enxebre**
Add ExpectAllMachinesLinkedToANode test
Needs https://github.com/openshift/machine-api-operator/pull/123

### CODE REVIEW

File: test/e2e/operator_expectations.go [hunk 1]
```diff
@@ -75,3 +77,47 @@ func ExpectOperatorStatusConditionDone() error {
...
+func ExpectAllMachinesLinkedToANode() error {
+	machineAnnotationKey := "machine"
+	listOptions := client.ListOptions{
+		Namespace: namespace,
+	}
+	machineList := capiv1alpha1.MachineList{}
+	nodeList := corev1.NodeList{}
+
+	err := wait.PollImmediate(1*time.Second, 1*time.Minute, func() (bool, error) {
+		if err := F.Client.List(context.TODO(), &listOptions, &machineList); err != nil {
```
Comment (spangenberg) [inline test/e2e/operator_expectations.go]: Why `context.TODO()` do you plan to change it in the future to an injected context?

File: test/e2e/operator_expectations.go [hunk 2]
```diff
@@ -75,3 +77,47 @@ func ExpectOperatorStatusConditionDone() error {
...
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	err = wait.PollImmediate(1*time.Second, 1*time.Minute, func() (bool, error) {
+		if err := F.Client.List(context.TODO(), &listOptions, &nodeList); err != nil {
```
Comment (spangenberg) [inline test/e2e/operator_expectations.go]: Same as above.

File: test/e2e/operator_expectations.go (hunk 1 above)
Comment (ingvagabund) [reply test/e2e/operator_expectations.go]: It has been done this way for a long time :-|

Comment (spangenberg) [reply test/e2e/operator_expectations.go]: Meh, we should rather use Background. IMHO `go vet` should mark that.

**Pull request merged**